lib/Sema/CSDiagnostics.cpp

```diff
@@ -3493,3 +3511,56 @@ bool MutatingMemberRefOnImmutableBase::diagnoseAsError() {
                             diagIDmember);
   return failure.diagnoseAsError();
 }
+
+bool InvalidTupleSplatWithSingleParameterFailure::diagnoseAsError() {
+  auto *anchor = getRawAnchor();
+
+  auto selectedOverload = getChoiceFor(anchor);
+  if (!selectedOverload || !selectedOverload->choice.isDecl())
+    return false;
+
+  auto *choice = selectedOverload->choice.getDecl();
+
+  auto *argExpr = getArgumentExprFor(anchor);
+  if (!argExpr)
+    return false;
+
+  using Substitution = std::pair<GenericTypeParamType *, Type>;
+  llvm::SmallDenseMap<GenericParamKey, Substitution, 8> substitutions;
```

Comment:
Hmm. The order of substitutions we print here is still going to be nondeterministic, because it's a hash table. Can you put these into a vector and sort on the `GenericParamKey` to make it deterministic?

Reply:
Specialization of DenseMapInfo here should make it deterministic because the hash is computed based on Depth + Index but I can make it a vector and sort it sure.

Reply:
SmallDenseMap doesn't have any order, though.

Reply:
For some reason I though that DenseMap would act more like a tree map...

Reply:
Done!